wavelength-plugins/splatoon.js

```diff
@@ -0,0 +1,248 @@
+				if (!this.can("ban", null, room)) return this.errorReply(`You must be a Room Moderator or higher to use this command.`);
+				if (room.id !== "splatoon") return this.errorReply(`This command only works in the Splatoon room.`);
+				if (!SPLATFEST.active) return this.errorReply(`Splatfest is not currently active.`);
+				SPLATFEST.active = false;
+				let splatfestUsers = Db.splatoon.keys();
+				for (let u in splatfestUsers) {
+					let splatProfile = Db.splatoon.get(splatfestUsers[u], {ranks: {}});
+					delete splatProfile.splatfest;
+					Db.splatoon.set(splatfestUsers[u], splatProfile);
+				}
+				if (Rooms("splatoon")) {
+					Rooms("splatoon").addRaw(`${WL.nameColor(user.name, true)} has disabled Splatfest.`);
+				}
+				// Clear splatfest teams
+				SPLATFEST.alpha = null;
+				SPLATFEST.bravo = null;
+			},
+
+			j: "join",
+			setteam: "join",
+			jointeam: "join",
+			join: function (target, room, user) {
+				if (!target) return this.parse(`/splatoonhelp`);
+				if (!SPLATFEST.active) return this.errorReply(`There is currently not a Splatfest. :(`);
+				if (!SPLATFEST.alpha === toId(target) || !SPLATFEST.bravo === toId(target)) return this.errorReply(`This is not a Splatfest team.`);
+				let splatProfile = Db.splatoon.get(user.userid);
+				splatProfile.splatfest = target;
+				Db.splatoon.set(user.userid, splatProfile);
+				return this.sendReply(`You have successfully joined Splatfest Team "${target}".`);
+			},
+
+			teams: "team",
+			team: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				if (!SPLATFEST.active) return this.errorReply(`There is currently not a Splatfest. :(`);
+				return this.sendReplyBox(`<strong>Splatfest Teams:</strong> ${SPLATFEST.alpha} and ${SPLATFEST.bravo}`);
+			},
+		},
+
+		name: "ign",
+		ingamename: "ign",
+		ign: function (target, room, user) {
+			if (!this.canTalk()) return false;
+			if (!target || target.length > 10) return this.errorReply(`Your IGN must be between 1-10 characters long.`);
+			let splatProfile = Db.splatoon.get(user.userid, {ranks: {}});
+			splatProfile.ign = target;
+			Db.splatoon.set(user.userid, splatProfile);
+			return this.sendReply(`Your In-Game Name has been set as: "${target}".`);
+		},
+
+		lvl: "level",
+		level: function (target, room, user) {
+			if (!this.canTalk()) return false;
+			if (!target || target > 99 || target < 1 || target.includes(".")) return this.errorReply(`Your level must be a number between 1-99 (no decimals).`);
+			let splatProfile = Db.splatoon.get(user.userid, {ranks: {}});
+			splatProfile.level = target;
+			Db.splatoon.set(user.userid, splatProfile);
+			return this.sendReply(`Your Level has been set to: Level ${target}.`);
+		},
+
+		profile: function (target, room, user) {
+			if (!this.runBroadcast()) return;
+			if (!target) target = user.userid;
+			target = toId(target);
+			if (target.length > 18) return this.errorReply("Usernames cannot exceed 18 characters.");
+			let self = this;
```

Comment:
No need for binding `this` to `self`; remove this variable declaration.

Reply:
Oh true because its no longer a function